pkg/network/protocols/postgres/telemetry.go

```diff
@@ -97,3 +98,131 @@ func (t *Telemetry) Count(tx *ebpf.EbpfEvent, eventWrapper *EventWrapper) {
+	pgMessagesCountBuckets [pgebpf.KerMsgCountNumBuckets]*libtelemetry.TLSAwareCounter
+	// pgMessagesCountNotFound		count cases when message was not detected in packet
+	pgMessagesCountNotFound *libtelemetry.TLSAwareCounter
+	// pgMessagesCountAboveMax		count cases when maximum number of messages was exceeded
+	pgMessagesCountAboveMax *libtelemetry.TLSAwareCounter
+	// pgMessagesCountErrors		count cases where an error occurred while counting messages
+	pgMessagesCountErrors *libtelemetry.TLSAwareCounter
+	// pgKernelMsgCountPlainPrev	save the last counters observed from the kernel, plain messages
+	pgKernelMsgCountPlainPrev pgebpf.PostgresKernelMsgCount
+	// pgKernelMsgCountPrev			save the last counters observed from the kernel, TLS messages
+	pgKernelMsgCountTlsPrev pgebpf.PostgresKernelMsgCount
+}
+
+// newPgKernelTelemetry this is the Postgres message counter store.
+func newPgKernelTelemetry() *PgKernelTelemetry {
+	metricGroup := libtelemetry.NewMetricGroup("usm.postgres_kernel", libtelemetry.OptStatsd)
+	pgKernelTel := &PgKernelTelemetry{
+		metricGroup:             metricGroup,
+		pgMessagesCountNotFound: libtelemetry.NewTLSAwareCounter(metricGroup, "messages_count_not_found"),
+		pgMessagesCountAboveMax: libtelemetry.NewTLSAwareCounter(metricGroup, "messages_count_above_max"),
+		pgMessagesCountErrors:   libtelemetry.NewTLSAwareCounter(metricGroup, "messages_count_errors"),
+	}
+	pgKernelTel.createMessagesCountBuckets(metricGroup)
+
+	return pgKernelTel
+}
+
+func (t *PgKernelTelemetry) createMessagesCountBuckets(metricGroup *libtelemetry.MetricGroup) {
+	for bucketIndex := range t.pgMessagesCountBuckets {
+		t.pgMessagesCountBuckets[bucketIndex] = libtelemetry.NewTLSAwareCounter(metricGroup,
+			"messages_count_bucket_"+(strconv.Itoa(bucketIndex+1)))
+	}
+}
+
+// update the Postgres message counter store with new counters from the kernel.
+// return if nothing to add, in this case also metric group summary ignores unmodified metrics.
+func (t *PgKernelTelemetry) update(kernMsgCounts *pgebpf.PostgresKernelMsgCount, isTLS bool) {
+	if kernMsgCounts == nil {
+		return
+	}
+	if isEmptyCounts(kernMsgCounts) {
+		log.Debugf("postgres kernel telemetry empty.")
+		return
+	}
+	delta := t.getDiff(kernMsgCounts, isTLS)
+	if isEmptyCounts(delta) {
+		log.Debugf("postgres kernel telemetry, no difference with previous counters.")
+		return
+	}
+	t.pgMessagesCountNotFound.Add(int64(delta.Messages_not_found), isTLS)
+	t.pgMessagesCountAboveMax.Add(int64(delta.Messages_above_max), isTLS)
+	t.pgMessagesCountErrors.Add(int64(delta.Messages_count_err), isTLS)
+	for bucketIndex := range t.pgMessagesCountBuckets {
+		t.pgMessagesCountBuckets[bucketIndex].Add(int64(delta.Messages_count_buckets[bucketIndex]), isTLS)
+	}
+	if isTLS {
+		t.pgKernelMsgCountTlsPrev = *kernMsgCounts
+	} else {
+		t.pgKernelMsgCountPlainPrev = *kernMsgCounts
+	}
+	s := t.metricGroup.Summary()
+	log.Debugf("postgres kernel telemetry, isTLS=%t, summary: %s", isTLS, s)
+}
```

Comment:
you are calculating a diff just to add it to a base, so why not "overriding the base?"
```
delta = new - base
base = base + delta
```

is equal to
```
base = new
```

Reply:
I want to avoid processing if there are no changes in counters

Reply:
but what processing do we have? the current implementation bears redundant processing

Reply:
https://github.com/DataDog/datadog-agent/pull/28503#discussion_r1722833170

Reply:
I took usage of "delta" from https://github.com/DataDog/datadog-agent/blob/main/pkg/network/protocols/http2/telemetry.go#L106 . I would like understand why it was used in first place and wheather indeed it is redundant

Reply:
I removed usage of "delta" in recent commit